### PR TITLE
Implement sum

### LINKED ISSUE #2: Add support for sum
Add support for sum using the [Kahan summation algorithm](https://en.wikipedia.org/wiki/Kahan_summation_algorithm).

```js
sum([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7]); // -> 15.3
```
**Updated README** under _Descriptive statistics_ and the table of contents.

### PR DESCRIPTION
**Pull request by mccraveiro**
closes #2

### CODE REVIEW

**Comment by boristane** [comment]
Thanks for your contribution @mccraveiro . Could you please fix the coding standards issues raised by Codacy ? I'll review afterwards.

File: README.md [hunk 1]
```diff
@@ -343,6 +344,16 @@ product([1, 2, 3, 4]); // -> 24
 
 If the data array is empty or contains a non finite `Number`, the method returns `undefined`.
 
+#### sum
+
+Return the sum of all entries in a numeric data array.
```
Comment (boristane) [inline README.md]: Please provide further details on the summation algorithm.

File: README.md [hunk 2]
```diff
@@ -343,6 +344,16 @@ product([1, 2, 3, 4]); // -> 24
 
 If the data array is empty or contains a non finite `Number`, the method returns `undefined`.
 
+#### sum
```
Comment (boristane) [inline README.md]: Please add to the table of contents after line `306`.

**Comment by mccraveiro** [comment]
Oh I haven't seen #10! It was kind of a race condition but @Binney was faster 😂

File: README.md [hunk 3]
```diff
@@ -343,6 +344,16 @@ product([1, 2, 3, 4]); // -> 24
...
+
+Return the sum of all entries in a numeric data array.
+
+```js
+sum([1, 2, 3, 4]); // -> 10
```
Comment (boristane) [inline README.md]: Please add an example where the floating point error is illustrated.

File: src/descriptive.js [hunk 4]
```diff
@@ -43,8 +43,28 @@ export function product(arr) {
   return result;
 }
 
+/**
+ * Returns the sum of all entries in a numeric data array.
+ * @param {Array} arr the data array
```
Comment (boristane) [inline src/descriptive.js]:
```suggestion
 * @param {Number[]} arr the data array
```

File: src/descriptive.js [hunk 5]
```diff
@@ -43,8 +43,28 @@ export function product(arr) {
...
+ * @returns {Number} the sum of all the data in the array
```
Comment (boristane) [inline src/descriptive.js]:
```suggestion
 * @returns {Number} the sum of all the elements of the data array
```

File: src/descriptive.js [hunk 6]
```diff
@@ -43,8 +43,28 @@ export function product(arr) {
   return result;
 }
 
+/**
+ * Returns the sum of all entries in a numeric data array.
```
Comment (boristane) [inline src/descriptive.js]: Please add details on the algorithm used, explaining why it is appropriate.

File: src/descriptive.js [hunk 7]
```diff
@@ -43,8 +43,28 @@ export function product(arr) {
...
+ * @returns {Number} the sum of all the data in the array
+ */
+export function sum(arr) {
+  if (!Array.isArray(arr) || arr.length === 0) return undefined;
+  let result = 0;
+  let compensation = 0;
+  for (let i = 0; i < arr.length; i += 1) {
+    if (!Number.isFinite(arr[i])) return undefined;
+    const compensatedInput = arr[i] - compensation;
+    const newResult = result + compensatedInput;
```
Comment (boristane) [inline src/descriptive.js]: The variable name `newResult` is ambiguous. Please change.

File: test/descriptive.test.js [hunk 8]
```diff
@@ -44,4 +45,15 @@ describe('Descriptive Statistics', () => {
     testUndefinedWithNullable(product);
   });
 
+  test('Summation', () => {
+    expect(sum([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7])).toBe(15.3);
```
Comment (boristane) [inline test/descriptive.test.js]: Please test at least two valid inputs. Also, I tend to test the case I add in the documentation. Could you please do the same ?

**Comment by mccraveiro** [comment]
@boristane Shouldn't we just close this PR as @Binney has already submitted another one?

**Pull request closed without merging**